From ea426d32eaa92310ec91c8363a8bc708b972c709 Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Wed, 18 Mar 2026 13:26:22 +0800
Subject: [PATCH] clk: aspeed: ast2700: fix clkgate register selection for clk
 enable

The clock enable path incorrectly writes all clock gate bits to
clkgate_clr1, even for clock IDs greater than 32. According to the
hardware design, clocks with ID > 32 should be controlled via
clkgate_clr2.

Fixes: 9267cfd3840 ("merge soc0/1 to be 1 file clk_ast2700")
Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: I636ab8381b5fc5d3b03aa5646cb5dd1f810eff64
(cherry picked from commit c596ac194419f51a96c9548b7f4ff5a90f11cff5)

Signed-off-by: santhoshs <santhoshs@ami.com>
---
 drivers/clk/aspeed/clk_ast2700.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/clk/aspeed/clk_ast2700.c b/drivers/clk/aspeed/clk_ast2700.c
index 97cc74b276..dc1cb92b8e 100644
--- a/drivers/clk/aspeed/clk_ast2700.c
+++ b/drivers/clk/aspeed/clk_ast2700.c
@@ -296,14 +296,11 @@ static int ast2700_soc1_clk_enable(struct clk *clk)
 {
 	struct ast2700_clk_priv *priv = dev_get_priv(clk->dev);
 	struct ast2700_scu1 *scu = (struct ast2700_scu1 *)priv->reg;
-	u32 clkgate_bit;
 
 	if (clk->id > 32)
-		clkgate_bit = BIT(clk->id - 32);
+		writel(BIT(clk->id - 32), &scu->clkgate_clr2);
 	else
-		clkgate_bit = BIT(clk->id);
-
-	writel(clkgate_bit, &scu->clkgate_clr1);
+		writel(BIT(clk->id), &scu->clkgate_clr1);
 
 	return 0;
 }
-- 
2.34.1

